MIT Press Journals.js

```diff
@@ -1,79 +1,123 @@
-		var link, title;
-		while ((link = links.iterateNext()) && (title = titles.iterateNext())) {
-			items[link.href] = Zotero.Utilities.trimInternal(title.textContent);
-		}
-		
-		Zotero.selectItems(items, function (items) {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
 				return true;
 			}
-			urls = new Array();
+			var articles = [];
 			for (var i in items) {
-				articles.push('http://www.mitpressjournals.org/doi/abs/' + getDOI(i));
+				articles.push(i);
 			}
-			ZU.processDocuments(articles, scrape)
+			ZU.processDocuments(articles, scrape);
 		});
 	} else {
-		scrape(doc, 'http://www.mitpressjournals.org/doi/abs/' + getDOI(url));
+		scrape(doc, url);
 	}
 }
 
+
 function scrape(doc, url){
-		if (doc.evaluate('//div[@class="abstractSection"]/p[contains(@class, "last") or contains(@class, "first")]', doc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
-			var abs = Zotero.Utilities.trimInternal(doc.evaluate('//div[@class="abstractSection"]/p[contains(@class, "last") or contains(@class, "first")]', doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent);
-		}
-		var doi = getDOI(doc.location.href);
-		var risurl = 'http://www.mitpressjournals.org/action/downloadCitation?doi=' + doi + '&include=cit&format=refman&direct=on&submit=Download+article+metadata';		
-		var pdfurl = doc.location.href.replace("/doi/abs/", "/doi/pdf/");
-		Zotero.Utilities.HTTP.doGet(risurl, function(text) {
-			var translator = Zotero.loadTranslator("import");
-			translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-			//Zotero.debug(text)
-			translator.setString(text);
-			translator.setHandler("itemDone", function(obj, item) {
-				//picks up some weird attachments and notes from the RIS - delete
-				item.attachments= [];
-				item.notes=[];
-				item.attachments.push({url:doc.location.href, title:item.publicationTitle + " Snapshot", mimeType:"text/html"})
-				item.attachments.push({url:pdfurl, title:item.publicationTitle + " Full Text PDF", mimeType:"application/pdf"});
-				if (abs) item.abstractNote = abs;
-				item.complete();	
+	var abs = text(doc, '.abstractSection');
+	var doi = getDOI(doc.location.href);
+	var risurl = 'http://www.mitpressjournals.org/action/downloadCitation?doi=' + doi + '&include=cit&format=refman&direct=on&submit=Download+article+metadata';
+	var pdfurl = doc.location.href.replace("/doi/abs/", "/doi/pdf/");
+	Zotero.Utilities.HTTP.doGet(risurl, function(text) {
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+		//Zotero.debug(text)
+		translator.setString(text);
+		translator.setHandler("itemDone", function(obj, item) {
+			//picks up some weird attachments and notes from the RIS - delete
+			item.attachments= [];
+			item.notes=[];
+			item.attachments.push({
+				url: doc.location.href,
```

Comment:
and here -- are we concerned that this is inconsistence, returning abstract or full text depending on where user is?